nav/virtual_layers/src/virtual_layer.cpp

```diff
@@ -0,0 +1,146 @@
+#include <virtual_layers/virtual_layer.h>
+#include <pluginlib/class_list_macros.h>
+
+#include <random>
+
+PLUGINLIB_EXPORT_CLASS(virtual_layers::VirtualLayer, costmap_2d::Layer)
+
+using costmap_2d::LETHAL_OBSTACLE;
+using costmap_2d::NO_INFORMATION;
+
+namespace virtual_layers
+{
+
+VirtualLayer::VirtualLayer() {
+  // ros::Time last_updated_ = ros::Time::now();
+  // ROS_INFO("Start");
+}
+
+void VirtualLayer::onInitialize()
+{
+  ros::NodeHandle nh("~/" + name_);
+  current_ = true;
+  default_value_ = NO_INFORMATION;
+  matchSize();
+
+  dsrv_ = new dynamic_reconfigure::Server<costmap_2d::GenericPluginConfig>(nh);
+  dynamic_reconfigure::Server<costmap_2d::GenericPluginConfig>::CallbackType cb = boost::bind(
+      &VirtualLayer::reconfigureCB, this, _1, _2);
+  dsrv_->setCallback(cb);
+}
+
+void VirtualLayer::matchSize()
+{
+  Costmap2D* master = layered_costmap_->getCostmap();
+  resizeMap(master->getSizeInCellsX(), master->getSizeInCellsY(), master->getResolution(),
+            master->getOriginX(), master->getOriginY());
+}
+
+
+void VirtualLayer::reconfigureCB(costmap_2d::GenericPluginConfig &config, uint32_t level)
+{
+  enabled_ = config.enabled;
+}
+
+std::vector<std::vector<double>> getRandomPoints(double t) {
+  // int n = 3;
+  // double point_x;
+  // double point_y;
+  std::vector<std::vector<double>> points = {{-1 - t , 1}, {-1 - t,-1}};
+  // std::random_device rd_x;
+  // std::random_device rd_y;
+  // std::mt19937 gen_x(rd_x());
+  // std::mt19937 gen_y(rd_y());
+  // std::uniform_real_distribution<double> rand_x(-4, 4);
+  // std::uniform_real_distribution<double> rand_y(-4, 4);
+
+  // for (int xy = 0; xy < n ; xy++){
+  //   point_x = rand_x(rd_x);
+  //   point_y = rand_y(rd_y);
+  //   points.push_back({point_x, point_y});
+  //   ROS_INFO("%f %f" , point_x, point_y);
+  // }
+
+  return points;
+}
+
+std::vector<std::vector<double>> pointsToLine(std::vector<std::vector<double>> points, double t) {
+  std::vector<std::vector<double>> coord;
+  coord.push_back({points[0][0], points[0][1]});
+  int m = points.size() - 1;
+  for (int j = 0; j < m; j++) {
```

Comment:
It's better to change the condition into:
`for (int j = 0; j < points.size() - 1; ++j)` and delete the m variable